src/App.tsx

```diff
@@ -0,0 +1,59 @@
+import { SearchBar } from "./SearchBar";
+import ResultsTable from "./ResultsTable";
+import data from "./data.json";
+import { useState } from "react";
+import Tabs from "./Tabs";
+
+export default function App() {
+  const [parkingSpots, setParkingSpots] = useState(getTenRandomParkingSpots());
+  const [selectedTab, setSelectedTab] = useState("Nearby");
+  const tabOptions = ["Nearby", "History", "Favourites"];
+
+  function submitSearch(location: string): void {
+    setParkingSpots(getTenRandomParkingSpots());
+  }
+
+  function getTenRandomParkingSpots(): ParkingData[] {
+    const shuffled = [...data.parkingSpots].sort(() => 0.5 - Math.random());
+    return shuffled.slice(0, 10);
+  }
```

Comment:
This function doesn't belong to this component, you probably should have a utility class/function with the parking operations. I think it's fine to keep here for now though and refactor when you integrate with a backend.